src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -68,21 +68,15 @@ internal bool SupportsSyntaxTree
             var treeSource = CreateLazyFullyParsedTree(
                 textSource,
                 info.Id.ProjectId,
-                GetSyntaxTreeFilePath(info),
+                GetSyntaxTreeFilePath(info.Attributes),
                 options,
                 language,
                 services);
 
-            // ownership of TextLoader information has moved to document state. clear out textloader the info is
-            // holding on. otherwise, these information will be held onto unnecesarily by documentInfo even after
-            // the info has changed by DocumentState.
-            // we hold onto the info so that we don't need to duplicate all information info already has in the state
-            info = info.WithTextLoader(null);
```

Comment:
so, I guess it is handled differently? otherwise, initial text will leak forever

Reply:
Yes (see later comment): we're now holding onto just Attributes, so we don't have to worry about it.